lib/roman_int/src/RomanInt.cpp

```diff
@@ -0,0 +1,235 @@
+    for (const auto n : roman_numerals_set) {
+      if (c == n) {
+        key = true;
+        break;
+      }
+    }
+
+    if (key == false) {
+      return false;
+    }
+  }
+
+  // Checks the order of the digits in the number.
+  const std::vector<std::pair<std::string_view, std::string_view>> pairs {
+    { "CM", " 900 " },
+    { "CD", " 400 " },
+    { "XC", " 90 " },
+    { "XL", " 40 " },
+    { "IX", " 9 " },
+    { "IV", " 4 " },
+    { "M",  " 1000 " },
+    { "D",  " 500 " },
+    { "C",  " 100 " },
+    { "L",  " 50 " },
+    { "X",  " 10 " },
+    { "V",  " 5 " },
+    { "I",  " 1 " }
+  };
+
+  for (const auto& pair : pairs) {
+    auto position = roman.find(pair.first);
+    while (position != std::string::npos) {
+      roman.erase(position, pair.first.length());
+      roman.insert(position, pair.second);
+      position = roman.find(pair.first);
+    }
+  }
+
+  std::stringstream ss;
+  ss << roman;
+  std::vector<std::int64_t> numbers;
+  for (std::int64_t number; ss >> number;) {
+    numbers.push_back(number);
+  }
+
+  for (std::size_t i = 1; i < numbers.size(); ++i) {
+    if (numbers.at(i-1) < numbers.at(i)) {
+      return false;
+    }
+  }
+
+  // If all checks are passed, returns true.
+  return true;
+}
+
+std::int64_t ToIntegerNumeral(const std::string_view roman_numeral) {
+  for (const auto& value : values) {
+    if (roman_numeral == value.first) {
+      return value.second;
+    }
+  }
+  // The next line is added so that the compiler does not cry.
+  // The program should not get here. Never...
+  return -1;
+}
+
+// Removes the subtraction rule.
+std::string ToSimpleForm(std::string complex_form) {
+  for (const auto& equivalent : equivalents) {
+    auto position = complex_form.find(equivalent.first);
+    if (position != std::string::npos) {
+      complex_form.erase(position, equivalent.first.length());
+      complex_form.insert(position, equivalent.second);
+    }
+  }
+  return complex_form;
+}
+
+// Adds the subtraction rule.
+std::string ToComplexForm(std::string simple_form) {
+  for (const auto& equivalent : equivalents) {
+    auto position = simple_form.find(equivalent.second);
+    if (position != std::string::npos) {
+      simple_form.erase(position, equivalent.second.length());
+      simple_form.insert(position, equivalent.first);
+    }
+  }
+  return simple_form;
+}
+
+std::int64_t ToInteger(std::string roman) {
+  auto simple_form = ToSimpleForm(roman);
+  std::vector<std::int64_t> integers(simple_form.size());
+  for (decltype(integers.size()) i{ 0 }; i < integers.size(); ++i) {
+    integers.at(i) = ToIntegerNumeral(std::string{ simple_form.at(i) });
+  }
+  return std::accumulate(integers.begin(), integers.end(), 0);
+}
+
+std::string ToRoman(std::int64_t integer) {
+  std::string roman;
+  for (const auto value : values) {
+    while (integer / value.second) {
+      integer -= value.second;
+      roman += value.first;
+    }
+  }
+  return ToComplexForm(roman);
```

Comment:
roman variable should be passed by r-value reference 
ToComplexForm(std::move(roman))